core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,571 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.evictor;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.collections.Pair;
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.ExceptionMessage;
+import alluxio.worker.WorkerContext;
+import alluxio.worker.block.BlockMetadataManagerView;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.allocator.Allocator;
+import alluxio.worker.block.meta.BlockMeta;
+import alluxio.worker.block.meta.StorageDirView;
+import alluxio.worker.block.meta.StorageTierView;
+
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class is used to evict blocks with LIRS policy on each tier. Different from LRU and LRFU,
+ * LIRS evicts the block with maximum Inter-Reference Recency (IRR). IRR is the number of distinct
+ * blocks between last access and second-to-last access to the block. Block with large IRR is
+ * predicted to be accessed again in a long time, so they needs to be evicted first to make room for
+ * other blocks which will be accessed soon. LIRS divides the tier into 2 parts: LIR cache and HIR
+ * cache. LIR cache holds blocks with small IRR and HIR cache holds blocks with large IRR. So blocks
+ * in HIR cache needs to be evicted first. If blocks in HIR cache gets a small IRR, they will be
+ * moved to LIR cache. And if the LIR cache is full, some blocks with relatively large IRR will be
+ * moved to HIR cache. LIRS can achieve better performance than LRU and LRFU in loop access pattern
+ * (like a1,a2,...,ak,a1,a2,...,ak,...). Besides, LIRS can achieve as good performance as LRU and

```

Comment:
This is a good candidate to be used as an example, could you write an explanation of running the algorithm on this example in this comment? Also, the documentation here is valuable to be put into the docs folder, otherwise, the user might not know whether he/she should choose this evictor.